CRM/Batch/Form/Entry.php

```diff
@@ -827,12 +792,11 @@ private function processMembership(&$params) {
               $campaignId = CRM_Utils_Array::value('campaign_id', $this->_values);
             }
           }
-          foreach (array('join_date', 'start_date', 'end_date') as $dateType) {
-            //CRM-18000 - ignore $dateType if its not explicitly passed
-            if (!empty($fDate[$dateType]) || !empty($fDate['membership_' . $dateType])) {
-              $formDates[$dateType] = CRM_Utils_Array::value($dateType, $value);
-            }
-          }
+
+          $formDates = array(
```

Comment:
all we need are these 2 values. for the renew function. It ignores join_date, prior to this pr it ignored start_date too. The value will only be set if passed in by the form now, since we have moved the unrelated date wrangling down.